pkg/spanconfig/spanconfigkvaccessor/validation_test.go

```diff
@@ -113,6 +120,65 @@ func TestValidateUpdateArgs(t *testing.T) {
 			},
 			expErr: "",
 		},
+		// Tests for system span configurations.
+		{
+			// Duplicate in toDelete.
+			toDelete: []spanconfig.Target{makeTenantClusterTarget(10), makeTenantClusterTarget(10)},
+			expErr:   "duplicate system targets .* in the same list",
+		},
+		{
+			// Duplicate in toUpsert.
+			toUpsert: []spanconfig.Record{
+				{
+					Target: makeTenantClusterTarget(10),
+				},
+				{
+					Target: makeTenantClusterTarget(10)},
+			},
+			expErr: "duplicate system targets .* in the same list",
+		},
+		{
+			// Duplicate in toDelete with some span targets.
+			toDelete: []spanconfig.Target{
+				spanconfig.MakeTargetFromSpan(roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("c")}),
+				makeTenantClusterTarget(roachpb.SystemTenantID.InternalValue),
+				spanconfig.MakeTargetFromSpan(roachpb.Span{Key: roachpb.Key("e"), EndKey: roachpb.Key("f")}),
+				makeTenantClusterTarget(roachpb.SystemTenantID.InternalValue),
+				spanconfig.MakeTargetFromSpan(roachpb.Span{Key: roachpb.Key("g"), EndKey: roachpb.Key("h")}),
+			},
+			expErr: "duplicate system targets .* in the same list",
+		},
+		{
+			// Duplicate in toDelete with some span targets.
+			toDelete: []spanconfig.Target{
+				spanconfig.MakeTargetFromSpan(roachpb.Span{Key: roachpb.Key("a"), EndKey: roachpb.Key("c")}),
+				makeTenantClusterTarget(roachpb.SystemTenantID.InternalValue),
```

Comment:
(Applies to earlier commits as well I think) Should we be using InternalValue? The name and surrounding comment suggests we shouldn't: https://github.com/cockroachdb/cockroach/blob/5a18c3ab8f4d1d096bc3a2d98bbe712d547c9cd1/pkg/roachpb/data.proto#L752-L753

Reply:
Could also make our helper take the TenantID type instead. If we got rid of type casting errors like suggested in the other review, some of this code could be simplified/inlined given there'd be no error handling needed.